src/sm/Basket.java

```diff
@@ -0,0 +1,12 @@
+package sm;
+
+import java.util.ArrayList;
+
+public class Basket {
+
+    ArrayList<Product> products = new ArrayList<Product>();
```

Comment:
make final and replace with interface List<Product> products = new ArrayList<Product>();